spec/services/csv/lettings_log_csv_service_spec.rb

```diff
@@ -1,430 +1,227 @@
 require "rails_helper"
 
 RSpec.describe Csv::LettingsLogCsvService do
-  context "when the user is support" do
-    let(:user) { create(:user, :support) }
-    let(:real_2021_2022_form) { Form.new("config/forms/2021_2022.json") }
+  let(:form_handler_mock) { instance_double(FormHandler) }
+  let(:organisation) { create(:organisation) }
+  let(:fixed_time) { Time.zone.local(2023, 6, 26) }
+  let(:log) do
+    create(
+      :lettings_log,
+      :completed,
+      startdate: fixed_time,
+      created_at: fixed_time,
+      updated_at: fixed_time,
+      mrcdate: fixed_time - 1.day,
+      voiddate: fixed_time - 2.days,
+      propcode: "ABCDEFG",
+      tenancycode: "HIJKLMN",
+      postcode_full: "NW9 5LL",
+      ppostcode_full: "TN23 6LZ",
+      created_by: user,
+      managing_organisation: organisation,
+    )
+  end
+  let(:user) { create(:user, :support, email: "s.port@jeemayle.com") }
+  let(:service) { described_class.new(user:, export_type:) }
+  let(:export_type) { "labels" }
+  let(:csv) { CSV.parse(service.prepare_csv(LettingsLog.where(id: logs.map(&:id)))) }
+  let(:logs) { [log] }
+  let(:headers) { csv.first }
+
+  it "calls the form handler to get all questions in order when initialized" do
+    allow(FormHandler).to receive(:instance).and_return(form_handler_mock)
+    allow(form_handler_mock).to receive(:ordered_lettings_questions_for_all_years).and_return([])
+    service
```

Comment:
Would it make sense to put this tests (and all the mocks) into context and add these lines into a before block, as we  don't seem to be mocking form in other tests (?)

Reply:
I'm not sure I see the benefit of this. The form handler needs mocking here for this test only, and then below where there is a context with multiple tests then I pulled the mocking needed there into a before for that collection of tests.